doc/fluid/design/dist_train/async_update.md

```diff
@@ -0,0 +1,52 @@
+# Design Doc: Asynchronous Update With Distributed Training
+
+## Background
+
+For the typical synchronous distributed training, some significant steps are as follows:
+
+1. A Trainer will compute the gradients and SEND them to the Parameter
+Server(PServer) nodes.
+1. After the PServer node received gradients came from all the Trainers, it would apply the gradient to the respective variables, and using an optimize algorithms(SGD,
+ Momentment...) to update the parameters.
+1. The Trainer would wait for the PServers finished the optimize stage, and GET the parameters from PServer, so all the Trainers would get the same parameters.
+
+In the synchronously distributed training, there should be a `Barrier` to synchronise the
+parameters after the optimizing stage. The performance of a distributed training job 
+depends on the lowest node, if there were hundreds or thousand training nodes in a Job,
+the performance of synchronously distributed training might be very slow because of 
+the slow node. So this design doc would introduce an approach to implement 
+*asynchronously* distributed training in PaddlePaddle Fluid.
+
+## Design
+
+<img src="./src/async_update.png" width="450"/>
+
+As the figure above, we describe a global view of asynchronously update process and use
+the parameter `w1` as an example to introduce the steps:
+1. For each gradient variables, they may distribute on different GPU card and aggregate
```

Comment:
Do you mean aggregate multiple gradients from gpus in a single machine? Have you considered to send gradients to pserver directly without aggregation? Each gpu is a independent training instance.

Reply:
> Do you mean aggregate multiple gradients from gpus in a single machine

Yes, because of the speed between different GPU cards is much faster than different nodes, so execute aggregation firstly and then send them may archive higher training speed.

But I think `Each gpu is a independent training instance ` is also a feasible way, maybe we implement it in future and do some experiments to compare the speed and accuracy.